exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ops/AllAnyOps.kt

```diff
@@ -0,0 +1,18 @@
+package org.jetbrains.exposed.sql.ops
+
+import org.jetbrains.exposed.sql.Op
+import org.jetbrains.exposed.sql.QueryBuilder
+import org.jetbrains.exposed.sql.UntypedAndUnsizedArrayColumnType
+
+class AllAnyOp<T>(val opName: String, val array: Array<T>) : Op<T>() {
+
+    override fun toQueryBuilder(queryBuilder: QueryBuilder) = queryBuilder {
```

Comment:
My suggestion would be to replace this with a sealed class, something like this:
```kt
sealed class AnyOrAllFromBaseOp<T>(
    val isAny: Boolean,
    val subSearch: Any
) : Op<T>() {
    override fun toQueryBuilder(queryBuilder: QueryBuilder): Unit = queryBuilder {
        if (isAny) {
            +"ANY ("
        } else {
            +"ALL ("
        }
        when (subSearch) {
            is Array<*> -> registerArgument(UntypedAndUnsizedArrayColumnType, subSearch)
        }
        +")"
    }
}

class AnyFromArrayOp<T>(array: Array<T>) : AnyOrAllFromBaseOp<T>(isAny = true, subSearch = array)

class AllFromArrayOp<T>(array: Array<T>) : AnyOrAllFromBaseOp<T>(isAny = false, subSearch = array)
```

With that, we'd be able to easily go in an extend support to subqueries, etc. by just adding a branch to the `when` block and some more subclasses.

Reply:
I refactored `opName` into `isAny` as you suggested but there might still be other row and array comparison operators, for example `SOME` though it's only an alias of `ANY`. (see [PostgreSQL docs](https://www.postgresql.org/docs/current/functions-comparisons.html#FUNCTIONS-COMPARISONS-ANY-SOME) and [MySQL docs](https://dev.mysql.com/doc/refman/8.0/en/any-in-some-subqueries.html)) If it's necessary to leave room for them (though I haven't find many cases supporting this), this class can be renamed to `RowAndArrayComparisonOp` as how PostgreSQL calls them and the old `opName` approach can be adopted.